### ISSUE TITLE
Support `defaultProp`

### ISSUE THREAD

**Issue by realStandal** [cristianbote/goober#409]
It'd be nice for the following to be typed and not throw an error in TypeScript:

```ts
import { styled } from 'goober'

type Props = {
  foo?: 'bar' | 'baz'
}

export Button = styled<'button', Props>('button')`
  ...
`

Button.defaultProps = {
  foo: 'bar'
}
```

Locally, I can add a `defaultProps` field to the [object intersected onto the `StyledVNode` type](https://github.com/cristianbote/goober/blob/bb7fd3e785e05496cf7ed4a1bded8f13946853a3/goober.d.ts#L72) and give it a type of `T`. This is when using React, I'm not sure if this is a React-only sort of thing and out of scope for goober.

It'd be a development-time only addition, at least haha :)

**Comment by realStandal** [comment]
Or even if `StyledVNode` were to be an interface so it can be overridden from user-land, much like how `DefaultTheme` is defined.

The functionality works in JavaScript, the types just break compilation.

**Comment by cristianbote** [comment]
Heyo! Any chance you can PR this?

**Comment by realStandal** [comment]
@cristianbote Of course, see the linked one above :)

**Comment by cristianbote** [comment]
published `2.1.3` with the above! Thank you!
